exporter/fileexporter/factory.go

```diff
@@ -155,6 +131,20 @@ func createLogsExporter(
 	)
 }
 
+func newFileExporter(conf *Config, writer io.WriteCloser) *fileExporter {
+	return &fileExporter{
+		path:             conf.Path,
+		formatType:       conf.FormatType,
+		file:             writer,
+		tracesMarshaler:  tracesMarshalers[conf.FormatType],
+		metricsMarshaler: metricsMarshalers[conf.FormatType],
+		logsMarshaler:    logsMarshalers[conf.FormatType],
```

Comment:
This is the important bit. Gotta instantiate all the marshalers just in case the same `conf` is used for different telemetry signals.

Reply:
Thank you for discovering this issue, please add a Changelog.